app/assets/javascripts/discourse/app/instance-initializers/page-tracking.js

```diff
@@ -10,6 +10,7 @@ export default {
 
   initialize(owner) {
     // Tell our AJAX system to track a page transition
+    // eslint-disable-next-line ember/no-private-routing-service
     const router = owner.lookup("router:main");
     router.on("routeWillChange", this.handleRouteWillChange);
```

Comment:
`routeWillChange` is available on the public router service, right? Could we use that here instead of the private router?

Reply:
The snag is that the router also passed into lib/page-tracker `startPageTracking()` - where it's used, among other things, to `send("refreshTitle")` - which is an action on a `DiscourseRoute`… 😩